app/components/layout.js

```diff
@@ -68,16 +66,22 @@ class Layout extends React.Component {
           </div>
         </div>
 
-        <Button onClick={this.connectToServer} value='Connect To Server' />
+        <Button onClick={this.connectToServer.bind(this)} value={this.props.connectionStatus} />
+
+        <Table />
       </div>
     )
   }
 }
 
-Layout.propTypes = {
-  // Validation logic
-};
+function mapDispatchToProps(dispatch) {
+  return {
+    actions: bindActionCreators(actions, dispatch)
+  }
+}
 
-window.onload = function(){
-  ReactDOM.render(<Layout/>, document.getElementById('layout-container'));
+function mapStateToProps(state) {
+  return state
 }
+
+export default connect(mapStateToProps, mapDispatchToProps)(Layout)
```

Comment:
I'm not sure I understood that. 
1. Whats wrong with exporting a derived value? 
2. Would removing the shorthand matter?
```
let connectedLayout = connect(mapStateToProps, mapDispatchToProps)(Layout)
export default connectedLayout
```

This is part of redux where in we connect a component to the store (single state tree), and define what part of the store it shall have access to.